include/highfive/H5File.hpp

```diff
@@ -55,6 +55,18 @@ class File: public Object, public NodeTraits<File>, public AnnotateTraits<File>
                   unsigned openFlags = ReadOnly,
                   const FileAccessProps& fileAccessProps = FileAccessProps::Default());
 
+    ///
+    /// \brief File
+    /// \param filename: filepath of the HDF5 file
+    /// \param openFlags: Open mode / flags ( ReadOnly, ReadWrite)
+    /// \param fileAccessProps: the file access properties
+    ///
+    /// Open or create a new HDF5 file
+    File(const std::string& filename,
+         unsigned openFlags,
+         const FileCreateProps& fileCreateProps,
+         const FileAccessProps& fileAccessProps = FileAccessProps::Default());
+
```

Comment:
SHould you instead add a new Prop to the ctor above?

Reply:
I'm worried about code such as:
```
auto fapl = FileAccessProps{};
File("foo.h5", File::ReadOnly, fapl);
```
if we simply change the signature to:
```
File(const std::string& filename,
     unsigned openFlags = ...,
     const FileCreateProps& fileCreateProps = FileCreateProps::Default(),
     const FileAccessProps& fileAccessProps = FileAccessProps::Default());
```
then that will break. OTOH, if we do:
```
File(const std::string& filename,
     unsigned openFlags = ...,
     const FileAccessProps& fileAccessProps = FileAccessProps::Default(),
     const FileCreateProps& fileCreateProps = FileCreateProps::Default());
```
then the order is different than the one for `H5Fcreate`.

Reply:
`createDataSet` also has first create, then access properties. So this works for me the best in terms of consistency.